### PR TITLE
fix(server): inject correct locker to request scope mutex

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
* **#6145** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6145?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6140** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6140?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6142** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6142?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/plugins/redis/mutex.ts [hunk 1]
```diff
@@ -31,66 +22,31 @@ else
...
-    const result = await this.redis.sendCommand(
-      new Command('EVAL', [unlockScript, '1', key, clientId])
-    );
-    if (result === 0) {
-      if (!ignoreUnlockFail) {
-        throw new Error(`Failed to release lock ${key}`);
-      } else {
-        this.logger.warn(`Failed to release lock ${key}`);
-      }
+        // TODO(@darksky): lock expired condition is not handled
```
Comment (forehalo) [inline packages/backend/server/src/plugins/redis/mutex.ts:43]: @darkskygit take a look

File: packages/backend/server/src/plugins/redis/mutex.ts [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-} from '../../fundamentals';
+import { ILocker, Lock } from '../../fundamentals';
+import { SessionRedis } from './instances';
 
 const lockScript = `local key = KEYS[1]
 local clientId = ARGV[1]
-local releaseTime = ARGV[2]
 
-if redis.call("get", key) == clientId or redis.call("set", key, clientId, "NX", "PX", releaseTime) then
+if redis.call("get", key) == clientId or redis.call("set", key, clientId, "NX", "EX", 60) then
```
Comment (darkskygit) [inline packages/backend/server/src/plugins/redis/mutex.ts:10]: is 60 seconds too long?
the expiration time here is used to clean up the lock when the server logic does not release the lock correctly due to an exception

File: packages/backend/server/src/plugins/redis/mutex.ts (hunk 1 above)
Comment (darkskygit) [reply packages/backend/server/src/plugins/redis/mutex.ts:43]: handled at next pr

File: packages/backend/server/src/plugins/redis/mutex.ts (hunk 2 above)
Comment (darkskygit) [reply packages/backend/server/src/plugins/redis/mutex.ts:10]: adjust to 500ms in next pr

Comment (forehalo) [reply packages/backend/server/src/plugins/redis/mutex.ts:10]: a functionality downtime with 60s is ok imo

Comment (darkskygit) [reply packages/backend/server/src/plugins/redis/mutex.ts:10]: reasonable, already revert to 60 seconds

**Review by darkskygit — APPROVED** [review]

**Pull request merged**